### ISSUE TITLE
Failed to launch the browser process || Syntax error

### ISSUE THREAD

**Issue by misterpyrrhuloxia** [jef/streetmerchant#2175]
## Expected Behavior

When I use the exact same `dotenv` on my Windows 10 machine, I have no issues. But when using it on my Raspberry Pi 3B, I get the below error.

## Current Behavior

```bash
[4:10:07 PM] error :: ✖ something bad happened, resetting streetmerchant in 5 seconds Failed to launch the browser process!
/home/pi/streetmerchant/node_modules/puppeteer/.local-chromium/linux-856583/chrome-linux/chrome: 1: /home/pi/streetmerchant/node_modules/puppeteer/.local-chromium/linux-856583/chrome-linux/chrome: Syntax error: "(" unexpected


TROUBLESHOOTING: https://github.com/puppeteer/puppeteer/blob/main/docs/troubleshooting.md
 {
  "stack": "Error: Failed to launch the browser process!\n/home/pi/streetmerchant/node_modules/puppeteer/.local-chromium/linux-856583/chrome-linux/chrome: 1: /home/pi/streetmerchant/node_modules/puppeteer/.local-chromium/linux-856583/chrome-linux/chrome: Syntax error: \"(\" unexpected\n\n\nTROUBLESHOOTING: https://github.com/puppeteer/puppeteer/blob/main/docs/troubleshooting.md\n\n    at onClose (/home/pi/streetmerchant/node_modules/puppeteer/lib/cjs/puppeteer/node/BrowserRunner.js:193:20)\n    at Interface.helper_js_1.helper.addEventListener (/home/pi/streetmerchant/node_modules/puppeteer/lib/cjs/puppeteer/node/BrowserRunner.js:183:68)\n    at Interface.emit (events.js:203:15)\n    at Interface.close (readline.js:397:8)\n    at Socket.onend (readline.js:173:10)\n    at Socket.emit (events.js:203:15)\n    at endReadableNT (_stream_readable.js:1145:12)\n    at process._tickCallback (internal/process/next_tick.js:63:19)"
}
```

## Steps to Reproduce

Happens just when running the script like normal

## Environment

- OS: Raspbian Stretch 9.13

```
# ** All configuration variables are optional **
# Copy this file and name it as `dotenv` (without backticks)
# Read https://jef.codes/streetmerchant/getting-started/#customization for help on customizing this file
########################################################################################################

ASCII_BANNER=
ASCII_COLOR=
AUTO_ADD_TO_CART=false
BROWSER_TRUSTED=
DESKTOP_NOTIFICATIONS=
DISCORD_NOTIFY_GROUP=
DISCORD_NOTIFY_GROUP_3060=
DISCORD_NOTIFY_GROUP_3060TI=
DISCORD_NOTIFY_GROUP_3070=
DISCORD_NOTIFY_GROUP_3080=
DISCORD_NOTIFY_GROUP_3090=
DISCORD_NOTIFY_GROUP_CORSAIR_SF=
DISCORD_NOTIFY_GROUP_RX6800=
DISCORD_NOTIFY_GROUP_RX6800XT=
DISCORD_NOTIFY_GROUP_RX6900XT=
DISCORD_NOTIFY_GROUP_RYZEN5600=
DISCORD_NOTIFY_GROUP_RYZEN5800=
DISCORD_NOTIFY_GROUP_RYZEN5900=
DISCORD_NOTIFY_GROUP_RYZEN5950=
DISCORD_NOTIFY_GROUP_SONYPS5C=
DISCORD_NOTIFY_GROUP_SONYPS5DE=
DISCORD_NOTIFY_GROUP_XBOXSX=
DISCORD_NOTIFY_GROUP_XBOXSS=
DISCORD_NOTIFY_GROUP_TEST=
DISCORD_WEB_HOOK=<REDACTED>
EMAIL_PASSWORD=<REDACTED>
EMAIL_TO=<REDACTED>
EMAIL_USERNAME=<REDACTED>
HEADLESS=
IN_STOCK_WAIT_TIME=
LOG_LEVEL=
LOW_BANDWIDTH=
MAX_PRICE_SERIES_3060=0
MAX_PRICE_SERIES_3060TI=0
MAX_PRICE_SERIES_3070=0
MAX_PRICE_SERIES_3080=900
MAX_PRICE_SERIES_3090=1800
MAX_PRICE_SERIES_CORSAIR_SF=0
MAX_PRICE_SERIES_RX6800=0
MAX_PRICE_SERIES_RX6800XT=0
MAX_PRICE_SERIES_RX6900XT=0
MAX_PRICE_SERIES_RYZEN5600=0
MAX_PRICE_SERIES_RYZEN5800=0
MAX_PRICE_SERIES_RYZEN5900=700
MAX_PRICE_SERIES_RYZEN5950=950
MAX_PRICE_SERIES_SONYPS5C=0
MAX_PRICE_SERIES_SONYPS5DE=0
MAX_PRICE_SERIES_XBOXSS=0
MAX_PRICE_SERIES_XBOXSX=0
MAX_PRICE_SERIES_TEST=0
MICROCENTER_LOCATION=
MQTT_BROKER_ADDRESS=
MQTT_BROKER_PORT=
MQTT_CLIENT_ID=
MQTT_PASSWORD=
MQTT_QOS=
MQTT_TOPIC=
MQTT_USERNAME=
OPEN_BROWSER=true
PAGE_BACKOFF_MIN=
PAGE_BACKOFF_MAX=
PAGE_SLEEP_MIN=
PAGE_SLEEP_MAX=
PAGE_TIMEOUT=
PAGERDUTY_INTEGRATION_KEY=
PAGERDUTY_SEVERITY=
PHILIPS_HUE_API_KEY=
PHILIPS_HUE_CLOUD_ACCESS_TOKEN=
PHILIPS_HUE_CLOUD_CLIENT_ID=
PHILIPS_HUE_CLOUD_CLIENT_SECRET=
PHILIPS_HUE_CLOUD_REFRESH_TOKEN=
PHILIPS_HUE_LAN_BRIDGE_IP=
PHILIPS_HUE_LIGHT_COLOR=
PHILIPS_HUE_LIGHT_IDS=
PHILIPS_HUE_LIGHT_PATTERN=
PHONE_CARRIER=<REDACTED>
PHONE_NUMBER=<REDACTED>
PLAY_SOUND=
PROXY_ADDRESS=
PROXY_PROTOCOL=
PROXY_PORT=
PUSHBULLET=<REDACTED>
PUSHOVER_EXPIRE=
PUSHOVER_RETRY=
PUSHOVER_TOKEN=
PUSHOVER_USER=
PUSHOVER_PRIORITY=
RESTART_TIME=
SCREENSHOT=true
SHOW_ONLY_BRANDS=
SHOW_ONLY_MODELS=
SHOW_ONLY_SERIES=3080,3090,ryzen5900,ryzen5950
SLACK_CHANNEL=
SLACK_TOKEN=
SMARTTHINGS_TOKEN=
SMARTTHINGS_SWITCH_LABEL=
SMTP_ADDRESS=
SMTP_PORT=
STORES=amazon,bestbuy,newegg,target,walmart:10000:30000,bandh:50000:60000
TELEGRAM_ACCESS_TOKEN=
TELEGRAM_CHAT_ID=
TWILIO_ACCOUNT_SID=
TWILIO_AUTH_TOKEN=
TWILIO_FROM_NUMBER=
TWILIO_TO_NUMBER=
TWITCH_ACCESS_TOKEN=
TWITCH_CHANNEL=
TWITCH_CLIENT_ID=
TWITCH_CLIENT_SECRET=
TWITCH_REFRESH_TOKEN=
TWITTER_ACCESS_TOKEN_KEY=
TWITTER_ACCESS_TOKEN_SECRET=
TWITTER_CONSUMER_KEY=
TWITTER_CONSUMER_SECRET=
TWITTER_TWEET_TAGS=
STREAMLABS_ACCESS_TOKEN=
STREAMLABS_TYPE=
STREAMLABS_IMAGE=
STREAMLABS_SOUND=
STREAMLABS_DURATION=
WEB_PORT=
```

**Comment by ZENinjaneer** [comment]
My puppeteer didn't come with "...node_modules/puppeteer/.local-chromium/linux-856583/chrome-linux/chrome" so I changed the index.ts in source to point to a chromium that I `npm install`'ed.

TLDR:
run `npm install chromium --save`
change index.ts main section like this:
```
  const chromium = require('chromium');
  browser = await launch({
    args,
    executablePath: chromium.path,
    defaultViewport: {
      height: config.page.height,
      width: config.page.width,
    },
    headless: config.browser.isHeadless,
  });
```
Then try to run it.

**Comment by misterpyrrhuloxia** [comment]
@ZENinjaneer, thanks for your help. I ran `npm install chromium --save` from the `streetmerchant` directory, modified my `~/streetmerchant/src/index.ts` file, and then ran `npm run start` and got the following error message.

```bash
[11:41:17 AM] error :: ✖ something bad happened, resetting streetmerchant in 5 seconds Failed to launch the browser process!
/home/pi/streetmerchant/node_modules/chromium/lib/chromium/chrome-linux/chrome: 1: /home/pi/streetmerchant/node_modules/chromium/lib/chromium/chrome-linux/chrome: cannot open 4��Q: No such file
/home/pi/streetmerchant/node_modules/chromium/lib/chromium/chrome-linux/chrome: 1: /home/pi/streetmerchant/node_modules/chromium/lib/chromium/chrome-linux/chrome: ELFL�: not found
/home/pi/streetmerchant/node_modules/chromium/lib/chromium/chrome-linux/chrome: 2: /home/pi/streetmerchant/node_modules/chromium/lib/chromium/chrome-linux/chrome: 4: not found
/home/pi/streetmerchant/node_modules/chromium/lib/chromium/chrome-linux/chrome: 3: /home/pi/streetmerchant/node_modules/chromium/lib/chromium/chrome-linux/chrome: Syntax error: word unexpected


TROUBLESHOOTING: https://github.com/puppeteer/puppeteer/blob/main/docs/troubleshooting.md
 {
  "stack": "Error: Failed to launch the browser process!\n/home/pi/streetmerchant/node_modules/chromium/lib/chromium/chrome-linux/chrome: 1: /home/pi/streetmerchant/node_modules/chromium/lib/chromium/chrome-linux/chrome: cannot open 4��Q: No such file\n/home/pi/streetmerchant/node_modules/chromium/lib/chromium/chrome-linux/chrome: 1: /home/pi/streetmerchant/node_modules/chromium/lib/chromium/chrome-linux/chrome: ELF\u0001\u0001\u0001\u0003\u0003\u0001L�: not found\n/home/pi/streetmerchant/node_modules/chromium/lib/chromium/chrome-linux/chrome: 2: /home/pi/streetmerchant/node_modules/chromium/lib/chromium/chrome-linux/chrome: 4: not found\n/home/pi/streetmerchant/node_modules/chromium/lib/chromium/chrome-linux/chrome: 3: /home/pi/streetmerchant/node_modules/chromium/lib/chromium/chrome-linux/chrome: Syntax error: word unexpected\n\n\nTROUBLESHOOTING: https://github.com/puppeteer/puppeteer/blob/main/docs/troubleshooting.md\n\n    at onClose (/home/pi/streetmerchant/node_modules/puppeteer/lib/cjs/puppeteer/node/BrowserRunner.js:193:20)\n    at Interface.helper_js_1.helper.addEventListener (/home/pi/streetmerchant/node_modules/puppeteer/lib/cjs/puppeteer/node/BrowserRunner.js:183:68)\n    at Interface.emit (events.js:203:15)\n    at Interface.close (readline.js:397:8)\n    at Socket.onend (readline.js:173:10)\n    at Socket.emit (events.js:203:15)\n    at endReadableNT (_stream_readable.js:1145:12)\n    at process._tickCallback (internal/process/next_tick.js:63:19)"
```

Maybe I made a mistake in how I modified the `index.ts` file? Here are the contents of my `index.ts` file as they are right now.

```bash
import * as Process from 'process';
import {config} from './config'; // Needs to be loaded first
import {startAPIServer, stopAPIServer} from './web';
import {Browser, launch} from 'puppeteer';
import {getSleepTime} from './util';
import {logger} from './logger';
import {storeList} from './store/model';
import {tryLookupAndLoop} from './store';

let browser: Browser | undefined;

async function sleep(ms: number) {
  return new Promise(resolve => setTimeout(resolve, ms));
}

/**
 * Schedules a restart of the bot
 */
async function restartMain() {
  if (config.restartTime > 0) {
    await sleep(config.restartTime);
    await stop();
    loopMain();
  }
}

/**
 * Starts the bot.
 */
async function main() {
  const args: string[] = [];

  // Skip Chromium Linux Sandbox
  // https://github.com/puppeteer/puppeteer/blob/main/docs/troubleshooting.md#setting-up-chrome-linux-sandbox
  if (config.browser.isTrusted) {
    args.push('--no-sandbox');
    args.push('--disable-setuid-sandbox');
  }

  // https://github.com/puppeteer/puppeteer/blob/main/docs/troubleshooting.md#tips
  // https://stackoverflow.com/questions/48230901/docker-alpine-with-node-js-and-chromium-headless-puppeter-failed-to-launch-c
  if (config.docker) {
    args.push('--disable-dev-shm-usage');
    args.push('--no-sandbox');
    args.push('--disable-setuid-sandbox');
    args.push('--headless');
    args.push('--disable-gpu');
  }

  // Add the address of the proxy server if defined
  if (config.proxy.address) {
    args.push(
      `--proxy-server=${config.proxy.protocol}://${config.proxy.address}:${config.proxy.port}`
    );
  }

  if (args.length > 0) {
    logger.info('ℹ puppeteer config: ', args);
  }

  await stop();
  const chromium = require('chromium');
  browser = await launch({
    args,
    executablePath: chromium.path,
    defaultViewport: {
      height: config.page.height,
      width: config.page.width,
      },
    headless: config.browser.isHeadless,
  });
  
  config.browser.userAgent = await browser.userAgent();

  for (const store of storeList.values()) {
    logger.debug('store links', {meta: {links: store.links}});
    if (store.setupAction !== undefined) {
      store.setupAction(browser);
    }

    setTimeout(tryLookupAndLoop, getSleepTime(store), browser, store);
  }

  await startAPIServer();
}

async function stop() {
  await stopAPIServer();

  if (browser) {
    // Use temporary swap variable to avoid any race condition
    const browserTemporary = browser;
    browser = undefined;
    await browserTemporary.close();
  }
}

async function stopAndExit() {
  await stop();
  Process.exit(0);
}

/**
 * Will continually run until user interferes.
 */
async function loopMain() {
  try {
    restartMain();
    await main();
  } catch (error: unknown) {
    logger.error(
      '✖ something bad happened, resetting streetmerchant in 5 seconds',
      error
    );
    setTimeout(loopMain, 5000);
  }
}

void loopMain();

process.on('SIGINT', stopAndExit);
process.on('SIGQUIT', stopAndExit);
process.on('SIGTERM', stopAndExit);
```
